### PR TITLE
[WIP] CLI-based docs and help

### PR DESCRIPTION
**Pull request by ignasi35**

### CODE REVIEW

File: bin/git-recent [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    exit $1
+}
+
+if [[ $# -ne 1 ]]; then
+        COMMIT_SUMMARY_LENGTH=12
+    else
+        COMMIT_SUMMARY_LENGTH=$1
+fi
+
+git log -"$COMMIT_SUMMARY_LENGTH" --graph --pretty=format:'%Cred%h%Creset %s%C(yellow)%d%Creset %Cgreen(%cr)%Creset [%an]' --abbrev-commit --date=relative
```
Comment (ignasi35) [inline bin/git-recent:21]: This is a completely new operation that some of the `usage()` functions will rely on to help users by pretty printing recent commits.

File: bin/oh-my-git [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    echo "   pr.open         Open a browser to the PR for the current repo and the provided <PR-id>"
+    echo "   pr.status       Print a status summary (merged, state, CI) for the givern <PR-id>"
+    echo "   pr.merged       Print wether the PR <PR-id> is already merged"
+    echo "   pr.new          Create and submit a new PR (only works if $EDITOR is E-macs or VSCode)"
+    echo 
+    echo "Git extensions:"
+    echo "   backport        Backports cherry-picked commits between the specified barnches"
+    echo "   clone-fork      Forks and prepares a working copy of the provided repo"
+    echo "   current-branch  Echoes the active branch in the working repo"
+    echo "   prepare         ????"
```
Comment (ignasi35) [inline bin/oh-my-git:42]: A few `????` around here.

File: git-functions.sh [hunk 3]
```diff
@@ -1,9 +1,11 @@
 
-function git.rebase.interative {
+function git.rebase.interactive {
```
Comment (ignasi35) [inline git-functions.sh:2]: typo

File: git-functions.sh [hunk 4]
```diff
@@ -43,7 +45,7 @@ function git.workbranch {
           copy-ide-settings ${REPO} $1          
           cd ../$1
         else 
-          echo "branch name must be passed, usage: git wbranch some-name"
+          echo "branch name must be passed, usage: git workbranch some-name"
```
Comment (ignasi35) [inline git-functions.sh:48]: This help was wrong 😅

File: README.md [hunk 5]
```diff
@@ -8,6 +8,7 @@ Some commands requires extra libraries, namely `hub` and `jq`.
 
 * `hub`  - https://hub.github.com/
 * `jq` - https://stedolan.github.io/jq/
+* `complete` (a default function in bash)
```
Comment (ignasi35) [inline README.md:11]:
```suggestion
* `complete`. That is a default function in `bash` but you must make sure `bashcompinit` is loaded if you are using `zsh`.
```

File: bin/oh-my-git [hunk 6]
```diff
@@ -0,0 +1,70 @@
+#!/bin/sh
```
Comment (ignasi35) [inline bin/oh-my-git:1]: This is a new command so users may `oh-my-git` anytime for a quick help.

File: bin/oh-my-git [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    case $1 in
+        -h|--help)
+            usage 0
+            ;;
+        *)
+            RESIDUAL_ARGS+=("$1")
+            ;;
+    esac
+    shift
+done
```
Comment (ignasi35) [inline bin/oh-my-git:70]: I considered supporting `oh-my-git <command> <arguments>` but I don't think it makes sense anymore. I'll probably remove these last few lines and keep `oh-my-git` very simple.

**Comment by ignasi35** [comment]
even with the `???` in the PR, I'd rather have this merged than keep it `WIP` forever.

**Review by octonato — APPROVED** [review]

**Comment by octonato** [comment]
Thanks @ignasi35

**Pull request merged**